File: languages/56/examples/test_problem.js
process.stdin.resume();
process.stdin.setEncoding('utf8');

var remainder = ''
process.stdin.on('data', function (chunk) {
	var lines = chunk.toString().split('\n');
	lines.unshift(remainder + lines.shift());
	remainder = lines.pop();
	lines.forEach(function(line) {
		if (line === '42') {
			process.exit();
		}
		process.stdout.write(line+'\n');
	});
});

File: languages/93/examples/test_problem.rs
use std::io::stdin;
use std::io::BufRead;
use std::io::BufReader;

fn main() {
    for line in BufReader::new(stdin()).lines() {
        match line {
            Ok(ref s) if s == "42" => break,
            Ok(ref s) => println!("{}", s),
            _ => continue
        }
    }
}

